File: langgraph/data/equiv/24.py
def find_greatest_divisor(number: int) -> int:
    divisor = number - 1
    while divisor > 0:
        if number % divisor == 0:
            return divisor
        divisor -= 1

# method name: largest_divisor -> find_greatest_divisor
# parameter name: n -> number
# changed for loop to while loop
# replaced reversed(range(n)) with explicit decrementing variable

File: langgraph/result/qwen/7b-eq2js/24.js
function findGreatestDivisor(number) {
    let divisor = number - 1;
    while (divisor > 0) {
        if (number % divisor === 0) {
            return divisor;
        }
        divisor--;
    }
}

// Example usage:
console.log(findGreatestDivisor(15)); // Output should be 5